src/GNDStk/XML/src/ctor.hpp

```diff
@@ -0,0 +1,73 @@
+
+// -----------------------------------------------------------------------------
+// XML Constructors
+// -----------------------------------------------------------------------------
+
+// default, move
+XML() = default;
+XML(XML &&) = default;
+
+
+// copy
+// Note: pugi::xml_document's is inaccessible
+XML(const XML &x)
+{
+   try {
+      convert(x,*this);
+   } catch (const std::exception &) {
+      log::context("XML(XML)");
+      throw;
+   }
+}
+
+
+// JSON
+explicit XML(const JSON &j)
+{
+   try {
+      convert(j,*this);
+   } catch (const std::exception &) {
+      log::context("XML(JSON)");
```

Comment:
My *context* messages are meant to give an intentionally *brief* "traceback" of where the error occurred. The primary error message would have occurred elsewhere. Then, XML(JSON) means class XML's constructor from a JSON object. (Which is basically how the constructor signature is written, excluding a const before JSON and reference after it.) Intentionally, a short context message suitable for someone who knows a bit of C++. A case could be made that context messages shouldn't be in the code at all, and should instead be the job of a supervisor code such as a debugger. Fair enough, but that presumes that a user doesn't mind the trouble of perhaps recompiling a code with debug symbols on, and then running the code again in a debugger. My thought was to product a brief but readable (if you know the system) traceback that someone would get, right away, along with an error that's thrown. That things like XML(JSON) mean XML's constructor from JSON would be documented somewhere, so that users would understand what they're seeing.